projections/weather/spec/weather/endpoint_provider_spec.rb

```diff
@@ -7,5 +7,32 @@
 module Weather
   describe EndpointProvider do
     subject { EndpointProvider.new }
+
+    context "Endpoint set" do
+      let(:expected) do
+        {"endpoint"=>{"url"=>"https://example.com"}}
+      end
+
+      it 'produces the expected output from the EndpointProvider' do
+        params = EndpointParameters.new(**{:endpoint=>"https://example.com"})
+        endpoint = subject.resolve_endpoint(params)
+        expect(endpoint.uri).to eq(expected['endpoint']['url'])
+        expect(endpoint.headers).to eq(expected['endpoint']['headers'] || {})
+        expect(endpoint.properties).to eq(expected['endpoint']['properties'] || {})
+      end
+    end
+
+    context "Endpoint not set" do
+      let(:expected) do
+        {"error"=>"Endpoint is not set - you must configure an endpoint."}
+      end
+
+      it 'produces the expected output from the EndpointProvider' do
+        params = EndpointParameters.new(**{})
+        expect do
+          subject.resolve_endpoint(params)
+        end.to raise_error(ArgumentError, expected['error'])
```

Comment:
Can't we just do `"Endpoint is not set - you must configure an endpoint."` instead of retrieving from the hash? Maybe I'm not seeing something very obvious here.  I'm thinking we are setting this up to populate more expectations in the future.

Reply:
We could do that - this was just what we did in V3 since it simplifies the code for the the let(:expected) bit.

Reply:
Disregard, ditto.